packages/prosess-vedtak/src/components/VedtakForm.jsx

```diff
@@ -1,437 +1,368 @@
 import styles from './vedtakForm.less';
-import VedtakOverstyrendeKnapp from './VedtakOverstyrendeKnapp';
 import BrevPanel from './brev/BrevPanel';
 import UstrukturerteDokumenter from './UstrukturerteDokumenter';
+import RevurderingPaneler from './revurdering/RevurderingPaneler';
+import redusertUtbetalingArsak from '../kodeverk/redusertUtbetalingArsak';
+import VedtakRevurderingSubmitPanel from './revurdering/VedtakRevurderingSubmitPanel';
+import VedtakSubmit from './VedtakSubmit';
 
 const isVedtakSubmission = true;
 
-const kanSendesTilGodkjenning = behandlingStatusKode =>
-  behandlingStatusKode === behandlingStatusCode.BEHANDLING_UTREDES;
+const fieldnames = {
+  SKAL_BRUKE_OVERSTYRENDE_FRITEKST_BREV: 'skalBrukeOverstyrendeFritekstBrev',
+  SKAL_HINDRE_UTSENDING_AV_BREV: 'skalHindreUtsendingAvBrev',
+  OVERSKRIFT: 'overskrift',
+  BRØDTEKST: 'brødtekst',
+  OVERSTYRT_MOTTAKER: 'overstyrtMottaker',
+  BEGRUNNELSE: 'begrunnelse',
+};
 
-const formName = 'VedtakForm';
+const transformRedusertUtbetalingÅrsaker = formikValues =>
+  Object.values(redusertUtbetalingArsak).filter(name =>
+    Object.keys(formikValues).some(key => key === name && formikValues[key]),
+  );
 
-export class VedtakForm extends Component {
-  constructor(props) {
-    super(props);
-    this.onToggleOverstyring = this.onToggleOverstyring.bind(this);
-    this.state = {
-      skalBrukeOverstyrendeFritekstBrev: props.skalBrukeOverstyrendeFritekstBrev,
-    };
-  }
+export const VedtakForm = ({
+  intl,
+  readOnly,
+  behandlingStatus,
+  behandlingresultat,
+  aksjonspunkter,
+  behandlingPaaVent,
+  vedtakVarsel,
+  previewCallback,
+  sprakkode,
+  ytelseTypeKode,
+  alleKodeverk,
+  personopplysninger,
+  arbeidsgiverOpplysningerPerId,
+  tilbakekrevingvalg,
+  vilkar,
+  tilgjengeligeVedtaksbrev,
+  informasjonsbehovVedtaksbrev,
+  dokumentdata,
+  submitCallback,
+  fritekstdokumenter,
+  lagreDokumentdata,
+  overlappendeYtelser,
+  revurderingsAarsakString,
+  resultatstruktur,
+  simuleringResultat,
+  resultatstrukturOriginalBehandling,
+  bgPeriodeMedAvslagsårsak,
+  medlemskapFom,
+  erRevurdering,
+  behandlingArsaker,
+}) => {
+  const [erSendtInnUtenArsaker, setErSendtInnUtenArsaker] = useState(false);
+  const onToggleOverstyring = (e, setFieldValue) => {
+    const kommendeVerdi = e.target.checked;
+    setFieldValue(fieldnames.SKAL_BRUKE_OVERSTYRENDE_FRITEKST_BREV, e.target.checked);
 
-  onToggleOverstyring() {
-    const { behandlingFormPrefix, clearFields: clearFormFields } = this.props;
-    const { skalBrukeOverstyrendeFritekstBrev } = this.state;
-    this.setState({
-      skalBrukeOverstyrendeFritekstBrev: !skalBrukeOverstyrendeFritekstBrev,
-    });
-    const fields = ['begrunnelse', 'overskrift', 'brødtekst'];
-    clearFormFields(`${behandlingFormPrefix}.VedtakForm`, false, false, ...fields);
-  }
+    if (kommendeVerdi) {
+      setFieldValue(fieldnames.SKAL_HINDRE_UTSENDING_AV_BREV, false);
```

Comment:
Jeg får feil her. Er det mulig at formik gjør en eller annen update underveis etter at den første verdien er satt, som ødelegger for denne?

Reply:
på hvilken måte får du feil?

Reply:
fikset